### DISCUSSION TITLE
Access Zitadel instance locally with externalDomain configured

### DISCUSSION THREAD

**Discussion by sbont** [zitadel/zitadel#6330] [Q&A] [answered]
I've deployed Zitadel on a Linux VPS and have set the external domain to my public IP, port 9000 so that I can reach it from outside.
Now, also running on my linux server is my backend service which needs to make calls to Zitadel. When trying to call localhost:9000 I get an "instance not found", which I guess has to do with "localhost" not matching the configured externalDomain. However, is it somehow possible to reach Zitadel over localhost? Or would I need to set up a reverse proxy for this use case?

**Comment by eliobischof** [comment, marked as answer]
If you send your externaldomain in the Host header of your requests, ZITADEL maps the instance using this domain and the instance not found error should be gone.

**Reply by Congee** [reply]
@eliobischof As soon as I was led to https://zitadel.com/docs/support/troubleshooting#instance-not-found, I had a guess that people who run zitadel behind a proxy will hit the same issue. And aha, a quick github search shows a bunch of related discussions/issues. Also it's a hiccup for people who need to change a domain hard coded to `ExternalDomain`.

I don't have an opinion but a question - is `ExternalDomain` absolutely necessary? Is it possible for zitadel to get rid of it?

**Reply by eliobischof** [reply]
ATM, only the ExternalDomain property controls  how virtual instances (or the first instance) is mapped using the Host header.

I think you are particularly referring to this issue 
https://github.com/zitadel/zitadel/issues/5395?

**Reply by Congee** [reply]
>  it is needed to send mails and notifications

That's a legit reason from your link, thanks
